integration-libs/segment-refs/root/http-interceptors/occ-segment-refs.interceptor.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable, inject, isDevMode } from '@angular/core';
+import { Observable } from 'rxjs';
+import { LoggerService, OccEndpointsService, WindowRef } from '@spartacus/core';
+import { SegmentRefsConfig } from '../config/segment-refs-config';
+
+@Injectable({ providedIn: 'root' })
+export class OccSegmentRefsInterceptor implements HttpInterceptor {
+  private segmentRefs?: string | null;
+  private requestHeader?: string;
+  protected readonly SEGMENT_REFS_KEY = 'segment-refs';
+
+  protected logger = inject(LoggerService);
+
+  constructor(
+    protected config: SegmentRefsConfig,
+    protected occEndpoints: OccEndpointsService,
+    protected winRef: WindowRef
+  ) {
+    const url = this.winRef.location.href ?? '';
+    const queryParams = new URLSearchParams(url.substring(url.indexOf('?')));
+    this.segmentRefs = queryParams.get('segmentrefs');
```

Comment:
For easier possible extensibility, let's move this query param name to a `protected readonly` property

Reply:
done